lib/govuk_content_schemas/frontend_schema_generator.rb

```diff
@@ -39,14 +39,15 @@ def internal?(property_name)
 
   def required_properties
     return [] unless @publisher_schema.schema.has_key?('required')
-    @publisher_schema.schema['required'].reject { |property_name| internal?(property_name) }
+    (@publisher_schema.schema['required'] - INTERNAL_PROPERTIES) + ['base_path']
```

Comment:
good optimisation! :+1:  this could be repeated in `frontend_properties` as well, and then we could get rid of the `internal?` method.